decidim-core/spec/lib/common_passwords_spec.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+module Decidim
+  describe CommonPasswords do
+    let(:subject) do
+      Class.new(described_class) do
+        def self.common_passwords_path
+          Rails.root.join("tmp/common-passwords.txt")
```

Comment:
I think this is not working as expected or at least there is some kind of problem in regarding this spec in my machine. 

The problem is that when I run this file, it changes the `lib/decidim/db/common-passwords.txt` file

```bash
cd decidim-core/
bundle exec rspec ./spec/lib/common_passwords_spec.rb
git diff
``` 

```diff
diff --git a/decidim-core/lib/decidim/db/common-passwords.txt b/decidim-core/lib/decidim/db/common-passwords.txt
index 437c8f893..b8f795dd3 100644
--- a/decidim-core/lib/decidim/db/common-passwords.txt
+++ b/decidim-core/lib/decidim/db/common-passwords.txt
@@ -1,128420 +1,4 @@
-qwertyuiop
+VJHT29061987
 1234567890
 q1w2e3r4t5
-1q2w3e4r5t
-4815162342
(...)
```

Reply:
@lahdeero I think the reason for this is that we are explicitly calling `Decidim::CommonPasswords.common_passwords_path` on these lines:

https://github.com/mainio/decidim/blob/a3d88ecef147794c9694c3db4027c08e2b68133f/decidim-core/lib/decidim/common_passwords.rb#L16
https://github.com/mainio/decidim/blob/a3d88ecef147794c9694c3db4027c08e2b68133f/decidim-core/lib/decidim/common_passwords.rb#L18
https://github.com/mainio/decidim/blob/a3d88ecef147794c9694c3db4027c08e2b68133f/decidim-core/lib/decidim/common_passwords.rb#L24

In the `#initialize` method you can call the class method by calling `self.class.common_passwords_path`.

In the class method `.update_passwords!` you can just remove the class reference and it should call the class method because you're already in the class context.

This way it won't call the original class methods in case you have overridden them in the spec.

Reply:
@andreslucena @ahukkanen Thanks for the information! Should be fixed now.